### PR TITLE
[V2 branch] ]WildFly S2I V2

### PR DESCRIPTION
**Pull request by jfdenise**
Initial commit for WildFly S2i V2.

* Renamed/reworked images yaml files.
* Removed dependency on wildfly-cekit-modules
* Removed all modules bound to previous design
* Introduced wildfly run module.
* Reworked wildfly-s2i module
* Reworked tests to use Bootable JAR Maven plugin (server mode) and vanilla WildFly (or WildFly cloud legacy galleon feature-pack).
* Server is installed by the builder image in JBOSS_HOME that points to /opt/server.
* Added new tests to increase coverage.
* Removed tools that were based on old design
* Removed templates and imagestreams
* Updated github actions. Still a dependency on jfdenise behave steps due to :https://github.com/cekit/behave-test-steps/pull/34 that will be merged soon. Still a dependency on jfdenise due to usage of https://github.com/jfdenise/clusterbench/tree/wildfly-s2i-v2. Commented helm and deploy to openshift, will update when we have charts for these new s2i images.
* Removed doc content, Doc will have to be re-worked.

The tests are expected to fail, we first need to merge this PR. They passed in fork: https://github.com/jfdenise/wildfly-s2i/actions/runs/983250943

### CODE REVIEW

File: test/test-app-ejb/pom.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0"
-    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (luck3y) [inline test/test-app-ejb/pom.xml:2]: minor: extra tab here?

File: test/test-app-extension/pom.xml [hunk 2]
```diff
@@ -85,16 +86,33 @@
             <id>openshift</id>
             <build>
                 <finalName>DemoApp</finalName>
-                <plugins>
+                                <plugins>
```
Comment (luck3y) [inline test/test-app-extension/pom.xml:89]: minor: some extra spaces here?

File: wildfly-builder-image/image.yaml [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
+      value: "Platform for building and running JEE applications on WildFly"
     - name: io.k8s.display-name
-      value: "WildFly $WILDFLY_VERSION"
+      value: "WildFly"
     - name: io.openshift.expose-services
-      value: "8080:http,8778:jolokia"
+      value: "8080:http"
     - name: io.openshift.tags
-      value: "builder,wildfly,wildfly24"
+      value: "builder,wildfly,wildfly"
```
Comment (luck3y) [inline wildfly-builder-image/image.yaml:15]: wildfly is duplicated here

File: test/test-app-ejb/pom.xml (hunk 1 above)
Comment (jfdenise) [reply test/test-app-ejb/pom.xml]: Thank-you. Fixed

File: test/test-app-extension/pom.xml (hunk 2 above)
Comment (jfdenise) [reply test/test-app-extension/pom.xml]: Thank-you. Fixed

File: wildfly-builder-image/image.yaml (hunk 3 above)
Comment (jfdenise) [reply wildfly-builder-image/image.yaml]: Thank-you. Fixed

File: wildfly-builder-image/image.yaml [hunk 4]
```diff
@@ -1,20 +1,26 @@
...
-name: &imgName "wildfly/wildfly-centos7"
+name: &imgName "wildfly/wildfly-s2i-jdk11"
 description: "The WildFly s2i image"
 version: &imgVersion "dev"
-from: "centos/s2i-base-centos7"
+from: "registry.access.redhat.com/ubi8/ubi-minimal"
 labels:
     - name: io.k8s.description
-      value: "Platform for building and running JEE applications on WildFly $WILDFLY_VERSION"
+      value: "Platform for building and running JEE applications on WildFly"
```
Comment (jmesnil) [inline wildfly-builder-image/image.yaml:9]: maybe: "Tooling image to build Enterprise Java applications with WildFly"

I'd like to demphasize the "and running" part of the builder image.
And Entreprise Java is more correct (WildFly now does Jakarta EE and MicroProfile)

File: wildfly-builder-image/image.yaml [hunk 5]
```diff
@@ -1,20 +1,26 @@
...
     - name: io.k8s.display-name
-      value: "WildFly $WILDFLY_VERSION"
+      value: "WildFly"
     - name: io.openshift.expose-services
-      value: "8080:http,8778:jolokia"
+      value: "8080:http"
     - name: io.openshift.tags
-      value: "builder,wildfly,wildfly24"
+      value: "builder,wildfly"
     - name: maintainer
```
Comment (jmesnil) [inline wildfly-builder-image/image.yaml:16]: @jfdenise do you want to be the maintainer or should we move to a community mailing list address (eg `wildfly-dev`)?

File: wildfly-runtime-image/image.yaml [hunk 6]
```diff
@@ -1,32 +1,31 @@
 schema_version: 1
 
-name: &imgName "wildfly/wildfly-runtime-centos7"
+name: &imgName "wildfly/wildfly-runtime-jre11"
 description: "The WildFly runtime image"
 version: &imgVersion "dev"
```
Comment (jmesnil) [inline wildfly-runtime-image/image.yaml:5]: is it accurate?

File: test/test-app-clustering/pom.xml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+                <finalName>SampleApp</finalName>
+                <plugins>
+                    <plugin>
+                        <groupId>org.wildfly.plugins</groupId>
+                        <artifactId>wildfly-jar-maven-plugin</artifactId>
+                        <version>6.0.0.Alpha1</version>
+                        <configuration>
+                            <!-- some tests check for the provisioned galleon layers -->
+                            <record-state>true</record-state>
+                            <feature-pack-location>org.wildfly:wildfly-cloud-legacy-galleon-pack:24.0.0.Final</feature-pack-location>
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply test/test-app-clustering/pom.xml:44]: I think the question you raise is, do we want the legacy cloud feature-pack to be in the universe (and benefit from channels)? I am not sure. This is a legacy thing that we should deprecate and remove at some point.

**Comment by jmesnil** [comment]
This will not be done with this PR but we have to figure out how we will document the new images.

As much as possible, I would like to deemphasize how to "build" these images (which only concerns a few developers in the WildFly community) and focus on how to "use" these images.

imho, the building part can be done in Wiki pages in the project.
The docs inside the project codebase can then focus on the user guide.

File: test/test-app-clustering/pom.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (jfdenise) [reply test/test-app-clustering/pom.xml]: Propertized.

File: wildfly-builder-image/image.yaml (hunk 4 above)
Comment (jfdenise) [reply wildfly-builder-image/image.yaml]: Good suggestion. Fixed.

File: wildfly-builder-image/image.yaml (hunk 5 above)
Comment (jfdenise) [reply wildfly-builder-image/image.yaml:16]: I set wildfly-dev

**Comment by jfdenise** [comment]
@luck3y  @jmesnil Any final concern with thisPR?

**Comment by luck3y** [comment]
@jfdenise LGTM, other than the CI failure.

**Review by luck3y — APPROVED** [review]

**Review by jmesnil — APPROVED** [review]

**Pull request merged**